src/main/java/org/cactoos/http/HtWire.java

```diff
@@ -57,12 +58,17 @@
      */
     private final int port;
 
+    /**
+     * Supplier of sockets.
+     */
+    private final BiFunction<String, Integer, Socket> supplier;
```

Comment:
@krzyk This `supplier` is unnecessary and doesn't help you to reach a goal.

Reply:
@SharplEr Hmm, I think it does, I need to check that the Socket is not called with invalid port.
Without it I would need to create somekind of http server that would listend on ports 80 and 443. Which would be difficult considering that those ports are restricted (and might not be free).
Using this BiFunction (or IoCheckedBiFun) right now I can validate my assumptions in a test.